src/main/java/com/google/sps/servlets/ActionsServlet.java

```diff
@@ -95,45 +56,20 @@ private String urlQuery(String basePath, Map<String, String> parameters) throws
     List<Project> projectsList = new ArrayList<>();
     for (int i = 0; i < projectsListJson.size(); i++) {
       Map projectJson = (Map) projectsListJson.get(i);
```

Comment:
Raw types like used here are generally unsafe to use and should be avoided. The [Gson documentation](https://javadoc.io/doc/com.google.code.gson/gson/latest/com.google.gson/com/google/gson/Gson.html) provides a way with which you can define parametrized types so as to avoid using raw types.

If you want to read more about raw types, there's this [StackOverflow question](https://stackoverflow.com/questions/2770321/what-is-a-raw-type-and-why-shouldnt-we-use-it) and [this tutorial](https://docs.oracle.com/javase/tutorial/java/generics/rawTypes.html) from Oracle